src/storage/app-state/slices/addressesSlice.ts

```diff
@@ -101,14 +105,35 @@ export const syncAllAddressesTransactionsNextPage = createAsyncThunk(
 
     const state = getState() as RootState
     const addresses = selectAllAddresses(state)
-    const nextPage = state.addresses.transactionsPageLoaded + 1
+    let nextPage = state.addresses.transactionsPageLoaded
+    let newTransactionsFound = false
+    let allTransactionsLoaded = state.addresses.allTransactionsLoaded
+    let transactions: Transaction[] = []
 
-    // NOTE: Explorer backend limits this query to 80 addresses
-    const results = await Promise.all(
-      chunk(addresses, 80).map((addressesChunk) => fetchAddressesTransactionsNextPage(addressesChunk, nextPage))
-    )
+    while (!allTransactionsLoaded && !newTransactionsFound) {
+      nextPage += 1
 
-    return { page: nextPage, transactions: results.flat() }
+      // NOTE: Explorer backend limits this query to 80 addresses
+      const results = await Promise.all(
+        chunk(addresses, 80).map((addressesChunk) => fetchAddressesTransactionsNextPage(addressesChunk, nextPage))
+      )
+
+      transactions = results.flat()
+
+      if (transactions.length === 0) {
+        allTransactionsLoaded = true
+        break
+      }
+
+      newTransactionsFound = addresses.some((address) => {
+        const transactionsOfAddress = getTransactionsOfAddress(transactions, address)
+        const newTxHashes = extractNewTransactionHashes(transactionsOfAddress, address.transactions)
+
+        return newTxHashes.length > 0
+      })
+    }
+
+    return { page: nextPage, transactions }
```

Comment:
Here is where I query the explorer until I get new transactions or until all transactions have loaded.